File: bone-business/tpa-sass-vue3/src/enums/rule/ValueTypeEnum.ts
/**
 * 值类型枚举
 */
export enum ValueTypeEnum {
  fixed = "fixed",
  dynamic = "dynamic",
}

/**
 * 值类型标签
 */
export const ValueTypeLabels: Record<ValueTypeEnum, string> = {
  [ValueTypeEnum.fixed]: "固定值",
  [ValueTypeEnum.dynamic]: "动态值",
};

/**
 * 值类型选项
 */
export const ValueTypeOptions = Object.values(ValueTypeEnum).map((type) => ({
  label: ValueTypeLabels[type],
  value: type,
}));

/**
 * 获取值类型标签
 */
export const getValueTypeLabel = (type: string | ValueTypeEnum) => {
  return ValueTypeLabels[type as ValueTypeEnum];
};

File: bone-business/tpa-saas/bone-lowcode/bone-lowcode-infra/src/main/java/com/bone/lowcode/infra/domain/valueobject/ValueTypeEnum.java
package com.bone.lowcode.infra.domain.valueobject;

import lombok.Getter;

//规则中的值类型枚举
@Getter
public enum ValueTypeEnum {
    FIXED("fixed", "固定值"),
    DYNAMIC("dynamic", "动态值");

    private final String value;
    private final String desc;

    ValueTypeEnum(String value, String desc) {
        this.value = value;
        this.desc = desc;
    }
}
